pkg/enactmentstatus/message.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright The Kubernetes NMState Authors.
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package enactmentstatus
+
+import (
+	"regexp"
+	"strings"
+)
+
+// The function strips disarranged parts of the error messages, so it is easier to read for the user
+func FormatErrorString(errorMessage string) string {
+	var sb strings.Builder
+	errorLines := strings.Split(errorMessage, "\n")
+	index := 0
+	if strings.Contains(errorLines[0], "error reconciling NodeNetworkConfigurationPolicy") {
+		sb.WriteString(errorLines[0] + "\n")
+		index++
+	}
+
+	re := regexp.MustCompile(`\s*(failed to execute.*'exit status \d+') '{2}\s'`)
+	matches := re.FindStringSubmatch(errorLines[index])
+	if len(matches) == 2 {
+		sb.WriteString(strings.TrimRight(matches[1], " ") + "\n")
+		index++
+	}
+	index = skipLines(&errorLines, index)
+
+	selectAndFormatLines(index, &errorLines, &sb)
+
+	return sb.String()
+}
+
+//Loops over all lines in error message and selects lines that should be kept
+func selectAndFormatLines(index int, errorLines *[]string, sb *strings.Builder) {
+	for index < len(*errorLines) {
+		formatCurrentLine(&(*errorLines)[index], sb)
+		index++
+		index = skipLines(errorLines, index)
+	}
+}
+
+//Simplifies a line that should be kept in the error message
+func formatCurrentLine(errorLine *string, sb *strings.Builder) {
+	lineSplitByColon := strings.Split(strings.TrimRight(*errorLine, " "), ": ")
+	indent := ""
+	for _, lineSection := range lineSplitByColon {
+		sb.WriteString(indent + lineSection + "\n")
+		indent = indent + "  "
+	}
+}
+
+func skipLines(lines *[]string, index int) int {
```

Comment:
nit: IIUC when passing a slice, go passes the slice header, which points to the underlying array.
So it should not be neccessary to pass a pointer to the slice here